### DISCUSSION TITLE
map_probabilities

### DISCUSSION THREAD

**Discussion by bexy-b** [MaartenGr/BERTopic#2137] [General]
Hi there,

I had a question about the `map_probabilities` function. It seems to only update the probabilities attribute if `len(probabilities.shape) == 2` but my question is when is this condition actually met?

As far as I'm aware the shape should always be `(n_docs, )` if running a "regular" topic model since the probabilities are given for the assigned topic per document only, or if you set `calculate_probabilities=True` then it'll be (n_docs, n_topics) right? 

I want to understand if/when the probabilities per document are updated, but as far as I understand it this would only be recalculated if you called the hdbscan model again. The only thing that changes is that you're saying document X with probability p is now mapped to topic Y instead of topic Z but doesn't the probability p stay the same?

**Comment by MaartenGr** [comment]
When you run `.shape` on an array, it will give you back a tuple that contains one or more values. So when you do `len(probabilities.shape)` it will actually tell you how many dimensions that array has instead of the exact dimensions. Therefore, when you have two dimensions (regardless of their size), `len(probabilities)` will be 2.
